src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +174,311 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::UI_SetAppIcon);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+  request_->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(mock_app_));
+
+  MockProtocolHandler mock_app_protocol_handler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillOnce(ReturnRef(mock_app_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_app_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_icon";
+  EXPECT_CALL(*mock_app_, policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  EXPECT_CALL(*mock_app_, app_id()).WillRepeatedly(Return(kAppId));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  request_->Run();
+
+  const std::string expected_sync_file_name = file_system::ConvertPathForURL(
+      kAppStorageFolder + "/" + kAppFolder + "/" + kSyncFileName);
+  const std::string actual_sync_file_name =
+      (*result)[am::strings::msg_params][am::strings::sync_file_name]
+               [am::strings::value].asString();
+  EXPECT_EQ(expected_sync_file_name, actual_sync_file_name);
+
+  const uint32_t expected_app_id = kAppId;
+  const uint32_t actual_app_id = static_cast<uint32_t>(
+      (*result)[am::strings::msg_params][am::strings::app_id].asInt());
+  EXPECT_EQ(expected_app_id, actual_app_id);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentApp_ApplicationNotRegistered) {
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_apis::Result::APPLICATION_NOT_REGISTERED),
+          _)).WillOnce(Return(true));
+
+  request_->Run();
+}
+
+TEST_F(SetAppIconRequestTest,
+       Run_SyncFileNameWithForbiddenSymbols_InvalidData) {
+  (*message_)[am::strings::msg_params][am::strings::sync_file_name] =
```

Comment:
@BSolonenko you should re-create command for this test case as you change `message_` object after `SetUp()`, so your `request_` will contain not actual message

Reply:
@AKalinich-Luxoft  This works because the message is sent by pointer

Reply:
@BSolonenko you are wrong and @AKalinich-Luxoft is right.

Reply:
@AByzhynar After deeper investigation there was find out that `SetAppIconRequest` holds shared pointer to already created message object so changing of message content inside unit test also changes content of the same message inside `SetAppIconRequest`. Thats why it is not necessary to create command object after `setUp()`.

Reply:
@BSolonenko Right. But request with this message was already created.

Reply:
@BSolonenko Also always invoke `Init` method before request->run

Reply:
@AByzhynar @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/f19ba1152d21b86c2dcb86e945f8e30dc3c321c8